### PR TITLE
Disconnect peers which have invalid port or address information

### PR DESCRIPTION
**Pull request by Elfoniok**
Signed-off-by: Lukasz Foniok <lukaszfoniok@gmail.com>

### CODE REVIEW

File: golem/network/p2p/peersession.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 logger = logging.getLogger(__name__)
 
-P2P_PROTOCOL_ID = 14
+P2P_PROTOCOL_ID = 24
```
Comment (jiivan) [inline golem/network/p2p/peersession.py]: ?

File: golem/network/p2p/peersession.py [hunk 2]
```diff
@@ -407,15 +408,17 @@ def _react_to_hello(self, msg):
             if solve_challenge and not self.verified:
                 self._solve_challenge(challenge, difficulty)
         else:
-            self.p2p_service.add_peer(self.key_id, self)
-            if solve_challenge:
-                self._solve_challenge(challenge, difficulty)
+            if not self.p2p_service.add_peer(self.key_id, self):
+                self.disconnect(PeerSession.DCRNotValid)
```
Comment (jiivan) [inline golem/network/p2p/peersession.py:412]: maybe `return` after `disconnect()`. In this way we can reduce indentation of the remaining of this method.

**Review by jiivan — APPROVED** [review]

File: golem/network/p2p/peersession.py [hunk 3]
```diff
@@ -407,15 +408,18 @@ def _react_to_hello(self, msg):
             if solve_challenge and not self.verified:
                 self._solve_challenge(challenge, difficulty)
         else:
-            self.p2p_service.add_peer(self.key_id, self)
-            if solve_challenge:
-                self._solve_challenge(challenge, difficulty)
+            if not self.p2p_service.add_peer(self.key_id, self):
+                self.disconnect(PeerSession.DCRNotValid)
+                return
             else:
```
Comment (maaktweluit) [inline golem/network/p2p/peersession.py:414]: `else` after a single `if` ending in  `return` doest not add anything

**Review by maaktweluit — CHANGES_REQUESTED** [review]
one small comment, besides LGTM

**Review by maaktweluit — APPROVED** [review]

**Pull request closed without merging**